### PR TITLE
Added RandomJoke screen

### PR DESCRIPTION
**Pull request by Agrebennicov**
**This is part 1 of Random Joke Screen**
**What needs to be done:**
1. Create proper `BaseViewModel` to reduce some boilerplate;
2. Add local saving feature;
3. Add sharing feature.

**What has been done:**

1. Added `MainActivity` composable, VM;
4. Added `Random` screen, VM, Service, Repo;
5. Minor updates on error screen to center it similar to `Loader`;
6. Added "Try Again" feature to Error Screen;
7. Added some DI injections.

**How it looks:**
For demo purposes network call is delayed by 2 seconds - (API is very fast)
https://user-images.githubusercontent.com/17549029/181023400-9c6b54a2-349c-4940-8ffd-77f1d708a6c2.mp4

### CODE REVIEW

**Comment by Erdboden** [comment]
🔥 🔥

File: app/src/main/java/com/agrebennicov/jetpackdemo/features/main/MainViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor() : ViewModel() {
+    private val _state = mutableStateOf(MainState())
+    internal val state: State<MainState> = _state
```
Comment (Erdboden) [inline app/src/main/java/com/agrebennicov/jetpackdemo/features/main/MainViewModel.kt:12]: think internal is irrelevant in this app

File: app/src/main/java/com/agrebennicov/jetpackdemo/common/util/Util.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.agrebennicov.jetpackdemo.common.util
+
+import androidx.compose.animation.*
+import androidx.compose.animation.core.tween
+
+/**
```
Comment (Erdboden) [inline app/src/main/java/com/agrebennicov/jetpackdemo/common/util/Util.kt:6]: this is weird 🤔

**Review by Erdboden — APPROVED** [review]

File: app/src/main/java/com/agrebennicov/jetpackdemo/features/main/MainViewModel.kt (hunk 1 above)
Comment (Agrebennicov) [reply app/src/main/java/com/agrebennicov/jetpackdemo/features/main/MainViewModel.kt:12]: Maybe we'll move it multi-module sometime :)

**Pull request merged**